src/script/api/script_timemode.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file script_timemode.cpp Implementation of ScriptTimeMode. */
+
+#include "../../stdafx.h"
+#include "script_timemode.hpp"
+
+#include "../../safeguards.h"
+
+ScriptTimeMode::ScriptTimeMode(bool calendar)
```

Comment:
Constructors for squirrel objects need a special handling, it should be like AsyncMode I think.

Reply:
It can't work like ScriptCompanyMode? That's what I copied for this application.

Reply:
I tested and it works, now I'm wondering why AsyncMode uses the complex way.